query_delete.go

```diff
@@ -163,6 +163,11 @@ func (q *DeleteQuery) Limit(n int) *DeleteQuery {
 //
 // To suppress the auto-generated RETURNING clause, use `Returning("NULL")`.
 func (q *DeleteQuery) Returning(query string, args ...interface{}) *DeleteQuery {
+	if !q.hasFeature(feature.DeleteReturning) {
+		q.err = errors.New("bun: returning is not supported for current dialect")
+		return q
+	}
+
```

Comment:
For dialects that do not support DELETE RETURNING.
Before: We remained slient on misuse and ignore it when append query.
After: We return error on misuse.

I dont think current solution is correct, but i dont think we should break it,
as that might casue potential failures in upper-level application.

Reply:
On the other hand, I was very discouraged when I tried to use `Returning()` with `DeleteQuery` and received an empty result. I propose adding a note to the method's documentation, something like: "If the dialect does not support DELETE RETURNING, an empty result will be returned without an error."

Reply:
I dont have a strong preference, let @vmihailenco decide.

Reply:
I think returning an error might be annoying in some cases, but way easier to debug...